SPECS/hdf5/hdf5.spec

```diff
@@ -0,0 +1,1054 @@
+%global macrosdir %(d=%{_rpmconfigdir}/macros.d; [ -d $d ] || d=%{_sysconfdir}/rpm; echo $d)
+# No more Java on i686
+%ifarch %{java_arches}
+%bcond_without java
+%else
+%bcond_with java
+%endif
+%global version_main %(echo %{version} | cut -d. -f-2)
+%global so_version 200
+%global with_mpich 0
+%global with_openmpi 0
+%if %{with_mpich}
+%global mpi_list mpich
+%endif
+%if %{with_openmpi}
+%global mpi_list %{?mpi_list} openmpi
+%endif
+Summary:        A general purpose library and file format for storing scientific data
+# Patch version?
+#global snaprel -beta
+Name:           hdf5
+Version:        1.12.1
+Release:        11%{?dist}
+License:        BSD
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://portal.hdfgroup.org/display/HDF5/HDF5
+Source0:        https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-%{version_main}/hdf5-%{version}/src/hdf5-%{version}.tar.bz2
+Source1:        h5comp
+Patch0:         hdf5-LD_LIBRARY_PATH.patch
+# Fix fortran build with gcc 12
+# https://github.com/HDFGroup/hdf5/pull/1412
+Patch1:         hdf5-gfortran12.patch
+# Fix java build
+Patch3:         hdf5-build.patch
+# Remove Fedora build flags from h5cc/h5c++/h5fc
+# https://bugzilla.redhat.com/show_bug.cgi?id=1794625
+Patch5:         hdf5-wrappers.patch
+# For patches/rpath
+BuildRequires:  automake
+BuildRequires:  gcc
+BuildRequires:  gcc-c++
+BuildRequires:  gcc-gfortran
+BuildRequires:  hostname
+BuildRequires:  krb5-devel
+BuildRequires:  libaec-devel
+BuildRequires:  libtool
+# Needed for mpi tests
+BuildRequires:  openssh-clients
+BuildRequires:  openssl-devel
+BuildRequires:  time
+BuildRequires:  zlib-devel
+%if %{with java}
+BuildRequires:  hamcrest
+BuildRequires:  java-devel
+BuildRequires:  javapackages-tools
+BuildRequires:  junit
+BuildRequires:  slf4j
+%else
+Obsoletes:      java-hdf5 < %{version}-%{release}
+%endif
+BuildRequires: krb5-devel
+BuildRequires: openssl-devel
+BuildRequires: time
+BuildRequires: zlib-devel
+BuildRequires: hostname
+# For patches/rpath
+BuildRequires: automake
+BuildRequires: libtool
+# Needed for mpi tests
+BuildRequires: openssh-clients
+BuildRequires: libaec-devel
+BuildRequires: gcc, gcc-c++
```

Comment:
We do one package per line and keep them in alphabetical order. You can lint the spec and that should do it for you.